### DISCUSSION TITLE
Using the notFound() function results in a flash effect

### DISCUSSION THREAD

**Discussion by christian-tchaikovsky** [vercel/next.js#66593] [App Router]
When calling `notFound()` using the `next-themes` library, a flash effect occurs (You can see how the light theme changes to a dark one after restarting the page)

Exactly the same behavior appears when a hydration error occurs, but here is the expected result 

Please tell me how normal is this behavior with `notFound()`? Has anyone encountered this? What is the reason for this behavior?

**Comment by nitlix** [comment]
I can confirm encountering this, bumping this so this gets reviewed (;

**Comment by maxess3** [comment]
I have the same issue
